src/com/gfredericks/test/chuck/clojure_test.cljc

```diff
@@ -48,33 +47,34 @@
 #?(:cljs
 (defmethod ct/report [::chuck-capture :error]
   [m]
-  (swap! *chuck-captured-reports* conj m)))
+  (swap! *chuck-captured-reports* conj (assoc m ::testing-contexts (:testing-contexts (ct/get-current-env))))))
 
 #?(:cljs
 (defmethod ct/report [::chuck-capture :fail]
   [m]
-  (swap! *chuck-captured-reports* conj m)))
+  (swap! *chuck-captured-reports* conj (assoc m ::testing-contexts (:testing-contexts (ct/get-current-env))))))
 
 #?(:cljs
 (defmethod ct/report [::chuck-capture :pass]
   [m]
-  (swap! *chuck-captured-reports* conj m)))
+  (swap! *chuck-captured-reports* conj (assoc m ::testing-contexts (:testing-contexts (ct/get-current-env))))))
 
 (defn capture-reports*
   [reports-atom f]
   #?(:clj
-     (binding [ct/report #(swap! reports-atom conj %)]
+     (binding [ct/report #(swap! reports-atom conj (assoc % ::testing-contexts ct/*testing-contexts*))]
        (f))
 
      :cljs
      (binding [*chuck-captured-reports* reports-atom
-               cljs.test/*current-env* (cljs.test/empty-env ::chuck-capture)]
+               ct/*current-env* (assoc (ct/empty-env ::chuck-capture)
+                                       :testing-contexts (:testing-contexts (ct/get-current-env)))]
        (f))))
 
 (defmacro capture-reports
   [& body]
   `(let [reports# (atom [])]
-     (capture-reports* reports# (fn [] ~@body))
+     (capture-reports* reports# (fn [] (do ~@body)))
```

Comment:
what does this accomplish?

Reply:
It's a very obscure bugfix that allows `body` to really be a sequence of programs. eg., if `body` was `({:pre [foo]} (inc 1))`, we'd trigger `fn` precondition syntax without the wrapping `do`. Related: https://clojure.atlassian.net/browse/CLJ-2573

Reply:
ohwow that's nuts